File: app/src/main/java/com/example/a18/path/evaluator/EvaluatorActivity.java
package com.example.a18.path.evaluator;

import android.animation.ObjectAnimator;
import android.animation.ValueAnimator;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import com.example.a18.path.R;

public class EvaluatorActivity extends AppCompatActivity implements View.OnClickListener {

    private TextView view;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_evaluator);

        view = findViewById(R.id.text);
        view.setOnClickListener(this);

    }

    @Override
    public void onClick(View v) {
        ValueAnimator animator = ObjectAnimator.ofObject(new TextEvaluator(3),new MoneyText(0),new MoneyText(999.554d));
        animator.setDuration(1000);
        animator.addUpdateListener(animation -> {
            MoneyText money = (MoneyText) animation.getAnimatedValue();
            CharSequence content =  money.getContent(62,36);

            view.setText(content);
        });
        animator.start();
    }
}

File: app/src/main/java/com/example/a18/path/evaluator/EvaluatorActivity.kt
package com.example.a18.path.evaluator

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.animation.DecelerateInterpolator
import android.widget.TextView
import com.example.a18.path.R

class EvaluatorActivity : AppCompatActivity(), View.OnClickListener {
    override fun onClick(v: View?) {

        IncreaseNumber.begin(textView, 9.55566)
            .apply {
                leftSize = 64
                rightSize = 32
                index = 2
                joints = { content, left, right ->
                    content.append("$").append(left).append("👌").append(right).append("%")
                }
            }
            .createAnimator()
            .setDuration(200000)
            .apply {
                interpolator = DecelerateInterpolator(3.0f)
            }
            .start()
    }

    private var textView: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_evaluator)

        textView = findViewById(R.id.text)
        findViewById<View>(R.id.button).setOnClickListener(this)
    }

}

